### PR TITLE
add switch to allow SystemVar Path add

### PR DESCRIPTION
**Pull request by andrihitz**
Adds a switch option /SystemVar to add it to the SystemVar Path aswell.

### CODE REVIEW

File: choco/tools/chocolateyinstall.ps1
Comment (naveen521kk) [inline choco/tools/chocolateyinstall.ps1:27]: If I understand the code correctly it adds `C:\Program Files\Scripts` to the system path. Is there any reason why `C:\Program Files\Scripts` needs to be added to the system path(TinyTex is installed to `C:\tools\TinyTeX` usually, where `C:\tools` can be customised)?

By the way, the installation can be added to the system path using the command `tlmgr path --w32mode=admin add` (I think this should be the default).

On the other hand, `SystemVar` isn't clear can you rename it to something else like `AddToSystemPath` instead?

Comment (andrihitz) [reply choco/tools/chocolateyinstall.ps1]: Thanks for your feedback, I accidentally copied the wrong code from my local files. Fixed it.

Does this mean that the installation should be touched again to add it to the system path?

A good point, I just renamed it to match your suggestion. Thanks

**Review by cderv — COMMENTED** [review]
Hi @andrihitz 

It is not quite clear to me which issue we are trying to solve here. 

`tlmgr` should be added to PATH using its own mechanism `tlmg path add` which we do
https://github.com/rstudio/tinytex-releases/blob/adc16f8f09594ce6d62ae13978ec143363363dd0/choco/tools/chocolateyinstall.ps1#L16-L18 

Is this not enough ? Is it not working ? 

as @naveen521kk said, there is some configuration that can be done to this command. It is shown here
https://www.tug.org/texlive/doc/tlmgr.html#path
In some case, the flag is not even needed 
> If the user has admin rights, and the option --w32mode is not given, the setting w32_multi_user determines the location (i.e., if it is on then the system path, otherwise the user path is changed).

Does chocolatey is executed in Admin mode when you install ? 

Anayway, If those configuration needs to be access we can make that happens. 

Here we are adding a second way to add to PATH and I am not quite sure if this is a good solution as I don't fully comprehend the issue. 

Can you explain what you are solving here ? Thank you.

**Comment by andrihitz** [comment]
Hi @cderv 

Thanks for your Feedback.

Well, the issue I have, by default, the path var is just installed at the user's path and not in the system path, even if the command prompt was launched as admin.

I misunderstood the note by @naveen521kk for installing the system path with the onboard command in tlmgr.
In the newest commits, I changed my PowerShell command to that command.

For my understanding, I wasn't able to get any information about the w32_multi_user setting and what I had to change to automatically add the system path, Can you explain where I can find this setting?

Thanks for your Support.

File: choco/tools/chocolateyinstall.ps1
Comment (cderv) [inline choco/tools/chocolateyinstall.ps1:28]: If we were to do that, I think we just need to add the flag in the previous line when the option is passed to chocolatey. 

https://github.com/rstudio/tinytex-releases/pull/24/files#diff-5a71f76f6ddf465237c7306d0e59eb56e11db5cfb7281b4a5a647306dda6efe5R19-R20

Here we are putting tlmgr twice in the PATH. First user, then system. It does not seem right.

**Comment by cderv** [comment]
> For my understanding, I wasn't able to get any information about the w32_multi_user setting and what I had to change to automatically add the system path, Can you explain where I can find this setting?

I think it is a conf for tlmgr. Probably something you can set with `tlmgr conf tlmgr w32_multi_user=1` or `tinytex::tlmgr_conf(c("tlmgr", "w32_multi_user=1")`. Donc for `tlmgr conf` is in the man page: https://www.tug.org/texlive/doc/tlmgr.html#conf
There are also some config file : https://www.tug.org/texlive/doc/tlmgr.html#CONFIGURATION-FILE-FOR-TLMGR

This is probably what is controlling the default behavior of `tlmgr`. 

Do you want to try one of this configuration file to see if that change something ? 

Other we could add the flag, but in the first call of `path add` and not add a second call IMO

File: choco/tools/chocolateyinstall.ps1
Comment (andrihitz) [reply choco/tools/chocolateyinstall.ps1]: combined the two lines, but I am not sure if this is the way to do it.
At least it works with the if operator.

**Comment by cderv** [comment]
The change looks good to me. Did you manage to build and test it yourself ? does it install as admin with this change ?
It looks it should

**Comment by andrihitz** [comment]
Yes, I packed it on my local machine and installed it with and without the parameter, and it worked as intended. The path var will be removed while uninstalling.

File: choco/tools/chocolateyuninstall.ps1
Comment (cderv) [inline choco/tools/chocolateyuninstall.ps1:7]: One last thing, I wonder if we need to do both there or if there is a way to know what was done at installation, and what to do when uninstalling ? 

Here we are removing both user and system while we would have done only one during install I believe. 

What do you think ?

Comment (andrihitz) [reply choco/tools/chocolateyuninstall.ps1]: After digging in the Powershell Env's, I found a solution to check the path.
The Test was successful, do you think that's a good solution?
I haven't found a way to reuse/get the Install parameters.

Comment (cderv) [reply choco/tools/chocolateyuninstall.ps1]: @naveen521kk do you have a thought on that ? you may have more experience on chocolatey. Thanks

Comment (naveen521kk) [reply choco/tools/chocolateyuninstall.ps1]: Looks like no that's not possible, atm.

Comment (cderv) [reply choco/tools/chocolateyuninstall.ps1]: One way would be to save the flag (or install type used) in a file with the installation. Would that be possible and good practice ? 

Also is there a place to open such feature request or at least put the idea down ? Or do they say it was not something in the logic of chocolatey ?

Comment (naveen521kk) [reply choco/tools/chocolateyuninstall.ps1]: > One way would be to save the flag (or install type used) in a file with the installation. Would that be possible and good practice ?

Yeah, it should be possible (using a registry key is also another option). Not sure if it's good practice or not.

> Also is there a place to open such feature request or at least put the idea down ? Or do they say it was not something in the logic of chocolatey ?

I'll ask them.

Comment (naveen521kk) [reply choco/tools/chocolateyuninstall.ps1]: > Also is there a place to open such feature request or at least put the idea down ? Or do they say it was not something in the logic of chocolatey ?

There's an open issue: https://github.com/chocolatey/choco/issues/1479, so they'll fix it in the future.

Comment (cderv) [reply choco/tools/chocolateyuninstall.ps1]: Thank you

Comment (andrihitz) [reply choco/tools/chocolateyuninstall.ps1]: Thanks a lot.
I guess until they release the feature to use the installation parameter, the solution with searching for the Variable in the paths should work fine.
If this is fine for you as well, I would appreciate it if this could be merged.

**Review by naveen521kk — APPROVED** [review]
small typo, other than that sgtm

**Review by cderv — APPROVED** [review]
Looks good to me too ! 

@yihui ok to merge ? 

Should we trigger a patch release of the chocolatey bundle ? Otherwise it will be available next month.

**Comment by yihui** [comment]
> @yihui ok to merge ?

Yes.

> Should we trigger a patch release of the chocolatey bundle ? Otherwise it will be available next month.

I don't have an opinion. It's up to @andrihitz (whether a release is needed right now).

**Comment by cderv** [comment]
I have just merged. @andrihitz can you wait for next month automatic release ? 

Otherwise, I'll see what I can do for a path release for this version.

**Comment by andrihitz** [comment]
Thanks for merging. We can wait for next month's automatic release. @cderv

**Pull request merged**
